juju/juju
# Remove block devices from state and wire up new block device service

Remove block devices from mongo and wire up the dqlite implementation.
Key components:
- delete state blockdevices.go
- remove state block devices from migration
- create skeleton dqlite machine records when model is imported
- wire up dqlite block device import / export
- replace storage BlockDevice struct with the one in domain/blockdevice (it was essentially a dup)
- move block device struct methods from storage to domain to live with the struct
- wire up the calls to the new block device service

Note: 2 tests in `cmd/jujud/agent` are skipped
- TestDiskManagerWorkerUpdatesState
- TestMachineWorkers

This is because we cannot yet "prime" the dqlite model database with data like we can mongo.

## QA steps

bootstrap a controller and add a model (I used aws)
`deploy postgresql --storage pgdata=ebs`
juju dump-model and see that block devices are included with machine in the yaml
`juju status --storage` to see that storage is attached

bootstrap a second controller
migrate the model above to the second controller
ensure migration succeeds

## Links

**Jira card:** JUJU-5311
